src/PlayerSkeleton.java

```diff
@@ -25,3 +26,204 @@ public static void main(String[] args) {
 	}
 	
 }
+
+class Heuristics {
+	private int lineCleared;
+	
+	private double weightOfLineCleared;
+	
+	private int numberOfHoles;
+	
+	private double weightOfNumberOfHoles;
+	
+	private int totalWeightOfHoles;
+	
+	private double weightOfTotalWeightOfHoles;
+	
+	private int sumOfAdjacentColumnHeightDifference;
+	
+	private double weightOfSumOfAdjacentColumnHeightDifference;
+	
+	private int landingHeight;
+	
+	private double weightOfLandingHeight;
+	
+	private int totalHeight;
+	
+	private double weightOfTotalHeight;
+	
+	private int rangeOfHeight;
+	
+	private double weightOfRangeOfHeight;
+	
+	
+	public Heuristics(State thisState, State nextState) {
+		this.lineCleared = nextState.getRowsCleared();
+		
+		this.numberOfHoles = 0;
+		this.totalWeightOfHoles = 0;
+		for(int column = 0; column < State.COLS; column++) {
+			for(int row = nextState.getTop()[column]-1; row >= 0; row--) {
+				if(nextState.getField()[row][column] == 0){
+					this.numberOfHoles++;
+					this.totalWeightOfHoles += (20 - row);
+				}
+			}
+		}	
+		
+		this.sumOfAdjacentColumnHeightDifference = 0;
+		for(int column = 1; column < State.COLS; column++) {
+			this.sumOfAdjacentColumnHeightDifference += Math.abs(nextState.getTop()[column]-nextState.getTop()[column-1]);
+		}
+		
+		//the height of bottom gird 
+		this.landingHeight = State.ROWS;
+		for(int column = 0; column < State.COLS; column++) {

```

Comment:
I am not really sure about this actually. how should landing height be calculated?